rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/40_synonyms_sets_get.yml

```diff
@@ -0,0 +1,95 @@
+setup:
+  - skip:
+      version: " - 8.8.99"
+      reason: Introduced in 8.9.0
+  - do:
+      synonyms.put:
+        synonyms_set: test-synonyms-3
+        body:
+          synonyms_set:
+            - synonyms: "hello, hi"
+            - synonyms: "goodbye, bye"
+  - do:
+      synonyms.put:
+        synonyms_set: test-synonyms-1
+        body:
+          synonyms_set:
+            - synonyms: "test, check"
+            - synonyms: "good, great"
+            - synonyms: "bad, evil"
+  - do:
+      synonyms.put:
+        synonyms_set: test-synonyms-2
+        body:
+          synonyms_set:
+            - synonyms: "pc, computer"
+
+---
+"List synonyms set":
+  - do:
+      synonyms_sets.get: { }
+
+  - match:
+      count: 3
+
+  - match:
+      results:
+        - synonyms_set: "test-synonyms-1"
+          count: 3
+        - synonyms_set: "test-synonyms-2"
+          count: 1
+        - synonyms_set: "test-synonyms-3"
+          count: 2
+
+---
+"Pagination - size":
+  - do:
+      synonyms_sets.get:
+        size: 2
```

Comment:
Let's also have a test where we create more than 10 synonyms_size and request > 10 to ensure we can return more than 10.